ClangSharpPInvokeGenerator/Program.cs

```diff
@@ -100,20 +105,36 @@ public static int Run(InvocationContext context)
                 foreach (var file in files)
                 {
                     var translationUnitError = CXTranslationUnit.Parse(createIndex, file, arr, Array.Empty<CXUnsavedFile>(), translationFlags, out CXTranslationUnit translationUnit);
+                    bool skipProcessing = false;
 
                     if (translationUnitError != CXErrorCode.CXError_Success)
                     {
-                        Console.WriteLine($"Error: '{translationUnitError}' for '{file}'.");
-                        var numDiagnostics = translationUnit.NumDiagnostics;
+                        Console.WriteLine($"Error: Parsing failed for '{file}' due to '{translationUnitError}'.");
+                        skipProcessing = true;
+                    }
+                    else if (translationUnit.NumDiagnostics != 0)
+                    {
+                        Console.WriteLine($"Diagnostics for '{file}':");
```

Comment:
A switch to not print warnings or to not skip on error could probably be provided in the future. For now though, clang and llvm generation both occur without any diagnostics and providing diagnostics for other files will help with resolving any codegen issues.